test/flatten_test.js

```diff
@@ -0,0 +1,80 @@
+goog.module('datalayerhelper.helper.testing.flatten');
+goog.setTestOnly();
+
+const {DataLayerHelper} = goog.require('helper');
+
+describe('The `flatten` method of helper', () => {
+  beforeEach(function() {
+    this.dataLayer = [{a: 1, b: {c: {d: 4}, e: 5}}];
+    this.dataLayer.push({f: 6});
+    this.dataLayer.push({g: 7});
+
+    this.helper = new DataLayerHelper(this.dataLayer);
+    this.dataLayer.push({g: 8, i: 9});
+    this.dataLayer.push({'b.c': 3});
+  });
+
+  it('compresses a complex history into 1 call', function() {
+    expectDataLayerEquals(/* expected= */ [
+      {a: 1, b: {c: {d: 4}, e: 5}},
+      {f: 6},
+      {g: 7},
+      {g: 8, i: 9},
+      {'b.c': 3},
+    ], this.dataLayer);
+
+    this.helper.flatten();
+
+    expectDataLayerEquals(/* expected= */ [{
+      a: 1,
+      b: {c: 3, e: 5},
+      f: 6,
+      g: 8,
+      i: 9,
+    }], this.dataLayer);
+  });
+
+  it('does nothing extra when called twice in a row', function() {
+    this.helper.flatten();
+    this.helper.flatten();
+
+    expectDataLayerEquals(/* expected= */ [
```

Comment:
For reference, a parameter name comment is only required when the argument's meaning is not obvious from context.  In this case, based on the name of the function and the other parameter named data layer, it feels obvious to me.  However, no harm in having them!

Style guide link: https://google.github.io/styleguide/jsguide.html#formatting-param-name-comments